Update overview/history with a basic first version (#1373)
plone/documentation: docs/overview/index.md

(overview-label)=

# Overview

Plone is a mature, secure, and user-friendly content management system (CMS).

> Plone lets non-technical people create and maintain information for a public website or an intranet using only a web browser. Plone was first released to the public on October 4, 2001.

Reviewer: ```suggestion
Plone is a content management system (CMS) that is well-known for its user-friendly interface and robust security features. 
With Plone, even non-technical users can easily create and manage the content for a public website or intranet using only a web browser. 
Plone's intuitive interface and comprehensive set of features make it a popular choice for businesses, governments, universities and any organization  who need a reliable and easy-to-use CMS.

Plone has a long history and has been trusted by users around the world since its initial release on October 4, 2001. 
Over the years, Plone has undergone many improvements and enhancements, making it a highly mature and stable CMS. 
Additionally, Plone is supported by a strong community of users and developers who contribute to its ongoing success.
```

Plone has the maturity, stability, and reliability of an application maintained by open source developers with decades of experience, while continually evolving and adapting to modern technology.

Lots of customizations can be made trough-the-web, such as creating content types, themes, workflows, and much more.
A full filesystem based development workflow is also possible.
Plone can be extended and used as a framework on which to build custom CMS-like solutions.

Plone works as a:

- Full-featured server-side rendered HTML CMS.
- React-based frontend for editing and viewing content, backed by a server with a REST API.

> - Headless CMS server with a REST API, allowing a developer to build a custom frontend with their chosen technology.

Reviewer: ```suggestion
Plone is a highly customizable content management system (CMS) that allows users to tailor their website to their specific needs. 
With Plone, users can easily create custom content types, themes, and workflows using a web-based interface. 
Additionally, Plone offers a full filesystem-based development workflow, allowing developers to build custom solutions on top of the Plone framework. 
This makes it possible to create CMS-like solutions that are tailored to the unique requirements of the user.

Plone can be used in a variety of ways, including 

- a React-based frontend for editing and viewing content, 
- as a full-featured server-side rendered HTML CMS, 
- or as a headless CMS server with a REST API. 
  This allows developers to build custom frontends using their technology of choice, providing maximum flexibility and versatility. 
  
Overall, Plone is a powerful and versatile CMS that can be customized to suit a wide range of needs.
```

- **Plone is open source.**
  Plone is licensed under the GNU General Public License, the same license used by Linux.
  This gives you the right to use Plone without a license fee, and to improve upon the product.

- **Plone is supported.**
  There are over two hundred active developers in the Plone Development Team around the world, and a multitude of companies that specialize in Plone development and support.

- **Plone is extensible.**
  There is a multitude of add-on products for Plone to add new features and content types.
  In addition, Plone can be scripted using web standard solutions and open source languages.

- **Plone is technology neutral.**
  Plone can interoperate with most relational database systems—both open source and commercial—and runs on a vast array of platforms, including Linux, Windows, macOS, and BSD.

(overview-4-label)=

## High Level Overview for Developers

Plone is a content management platform with its backend written in Python.
The backend builds upon Zope, an open source web application server and development system, and thus on the pluggable Zope Component Architecture (ZCA).

> The backend has up until version 6 served HTML based content, with advanced resource management on the server to add and bundle CSS and javascript.

Reviewer: ```suggestion
Plone is a content management platform with its backend written in Python.
It is built on top of the open-source Zope web application server and development system. 
Plone is making use of the pluggable Zope Component Architecture (ZCA) to provide a highly modular and extensible system.

Up until version 6, Plone has used server-side rendering to generate HTML-based content, with advanced resource management features for adding and bundling CSS and JavaScript. 
Additionally, Plone's use of the ZCA makes it easy to extend and customize, allowing users to create unique, feature-rich websites that are tailored to their specific needs.
```

> With the release of Plone 6 there are now two out of the box supported configurations possible for a new Plone website.

Reviewer: ```suggestion
With the release of Plone 6, you now have the option to choose from two different out-of-the-box supported configurations when setting up a new Plone website.
```

You can still use the Python based backend server to render the content server side and deliver html to the browser.
This setup is referred to in the documentation as 'Classic UI' and has been supported by Plone since its release.

> For container based deployment you only need the plone-backend image, or a derivation with your customisations added.

Reviewer: ```suggestion
The Python-based backend server in Plone alone can still be used to render content server-side and deliver HTML to the browser, a setup that is referred to in the Plone documentation as "Classic UI". 
This configuration has been supported by Plone since its initial release and is still available in the latest versions of the platform. 
For container-based deployment, only the plone-backend image is required, or a derivative image with customizations added. 
```

The default and advised configuration for new websites in Plone is to use our new React Based javascript frontend called 'Volto'.
For this setup you still need to run the Python based backend server, but with the REST API enabled and an updated configuration profile.
In addition a separate NodeJS based frontend server will serve the javascript frontend resources and provide SSR with hydration.

> To deploy this setup using containers you will need the plone-frontend image of the frontend server.

Reviewer: ```suggestion
Plone's default and advised configuration for new websites is to use its new React-based JavaScript frontend called "Volto". 
This setup still requires running the Python-based backend server, but with the REST API enabled and an updated configuration profile. 
In addition, a separate Node.js-based frontend server is required to serve the JavaScript frontend resources and provide server-side rendering (SSR) with hydration. 
To deploy this setup using containers, the plone-frontend image of the frontend server is required.
```

Plone 6 is the version Plone where we support two programming language stacks, one for Python and Javascript.
The documentation has been rewritten, but for this first release you will find some repetition of concepts in the documentation structure.
For example for the development setup and information on deployment options.

> It will take some time before we find and can implement the best structure to explain these new possibilities and expansion of Plone its capabilities.

Reviewer: ```suggestion
The documentation for Plone 6 has been rewritten to reflect these changes, but some repetition of concepts can still be found in the structure of the documentation. 
For example, information on the development setup and deployment options may be repeated in multiple places. 
It will take some time before the best structure is found and implemented to explain the new possibilities and expanded capabilities of Plone. 
```

(overview-label)=

# Overview

Plone is a content management system (CMS) that is well-known for its user-friendly interface and robust security features. 
With Plone, even non-technical users can easily create and manage the content for a public website or intranet using only a web browser. 

> Plone's intuitive interface and comprehensive set of features make it a popular choice for businesses, governments, universities and any organization  who need a reliable and easy-to-use CMS.

Reviewer 2: ```suggestion
Plone's intuitive interface and comprehensive set of features make it a popular choice for businesses, governments, universities, and any organization that needs a reliable and easy-to-use CMS.
```

Plone has a long history and has been trusted by users around the world since its initial release on October 4, 2001. 
Over the years, Plone has undergone many improvements and enhancements, making it a highly mature and stable CMS. 
Additionally, Plone is supported by a strong community of users and developers who contribute to its ongoing success.

Plone has the maturity, stability, and reliability of an application maintained by open source developers with decades of experience, while continually evolving and adapting to modern technology.

> Lots of customizations can be made trough-the-web, such as creating content types, themes, workflows, and much more.

Reviewer 3: ```suggestion
Lots of customizations can be made through-the-web, such as creating content types, themes, workflows, and much more.
```

Note: "through-the-web" is jargon that I haven't heard much outside of the Plone community. It might be better to say "without writing code".

Author: Wild guess, the Zope comunity might have invented this jargon because they were one of the first allowing this kind of editing at all.  What could we improve?  

"Lots of customizations can be made through your browser, such as..."
"Lots of customizations can be done in your browser and are available as setings, such as ..."
"Lots of customizations can be done directly in the Plone Control-Panels, such as ..."

Reviewer 2: Here's another stab at it:

Many customizations can be made through your web browser—and do not require knowledge of how to write code—such as...

> A full filesystem based development workflow is also possible.

Reviewer 2: ```suggestion
A full file system based development workflow is also possible.
```

> A full file system based development workflow is also possible.

Reviewer 3: maybe source code repository-based?

Author: Yes, I first wanted to use VCS or version control system here, but that is <15 years terminology when we first struggled with CVS/VCS. 
Suggestion:

"A full file system based development workflow is possible and recommended for team work and deployment, backed by source code repositories. "

Reviewer 2: @fredvd I think you nailed it with that one.

Author: Will add the last one in a separate commit, thanks for the feedback!

Plone can be extended and used as a framework on which to build custom CMS-like solutions.

Plone works as a:

- Full-featured server-side rendered HTML CMS.
- React-based frontend for editing and viewing content, backed by a server with a REST API.
- Headless CMS server with a REST API, allowing a developer to build a custom frontend with their chosen technology.

(overview-3-label)=

## Key benefits

Security is built into Plone's architecture from the ground up.
Plone offers fine-grained permission control over content and actions.

> Plone is easy to set up compared to other CMS'es in its category, extremely flexible, and provides you with a system for managing web content that is ideal for project groups, communities, websites, extranets, and intranets.

Reviewer 2: ```suggestion
Plone is easy to set up compared to other CMSs in its category, extremely flexible, and provides you with a system for managing web content that is ideal for project groups, communities, websites, extranets, and intranets.
```

  The Plone Team includes usability experts who have made Plone easy and attractive for content managers to add, update, and maintain content.

- **Plone is international.**
  The Plone interface has more than 35 translations, and tools exist for managing multilingual content.

- **Plone follows standards and is inclusive.**
  Plone carefully follows standards for usability and accessibility.
  Plone is compliant with WCAG 2.1 level AA and aims for ATAG 2.0 level AA.

- **Plone is open source.**
  Plone is licensed under the GNU General Public License, the same license used by Linux.
  This gives you the right to use Plone without a license fee, and to improve upon the product.

- **Plone is supported.**
  There are over two hundred active developers in the Plone Development Team around the world, and a multitude of companies that specialize in Plone development and support.

- **Plone is extensible.**
  There is a multitude of add-on products for Plone to add new features and content types.
  In addition, Plone can be scripted using web standard solutions and open source languages.

- **Plone is technology neutral.**

>   Plone can interoperate with most relational database systems—both open source and commercial—and runs on a vast array of platforms, including Linux, Windows, macOS, and BSD.

Reviewer 3: "vast array" seems like hyperbole a bit

```suggestion
  Plone can interoperate with most relational database systems—both open source and commercial—and runs on many platforms, including Linux, Windows, macOS, and BSD.
```

(overview-4-label)=

## High Level Overview for Developers

Plone is a content management platform with its backend written in Python.

> It is built on top of the open-source Zope web application server and development system. 

Reviewer 2: ```suggestion
It is built on top of the open source Zope web application server and development system. 
```

> Plone is making use of the pluggable Zope Component Architecture (ZCA) to provide a highly modular and extensible system.

Reviewer 3: ```suggestion
Plone makes use of the pluggable Zope Component Architecture (ZCA) to provide a highly modular and extensible system.
```

(It wasn't wrong before, but this way reads slightly more natural to me as a native English speaker.)

Reviewer 2: Another suggestion, I think this needs to be added to the glossary, and marked up here.

## `docs/overview/index.md`

```
{term}`Zope Component Architecture`
```

## `glossary.md`

````md
ZCA
Zope Component Architecture
    The Python packages `zope.component` and `zope.interface` together provide facilities for defining, registering and looking up components.
    There are two basic kinds of components: adapters and utilities.

    Utilities are just components that provide an interface and that are looked up by an interface and a name.

    Adapters are components that are computed from other components to adapt them to some interface.
    Because they are computed from other objects, they are provided as factories, usually classes.

    ```{seealso}
    [Zope Component Architecture](https://zopecomponent.readthedocs.io/en/latest/narr.html)
    ```
````

It is also important find the right balance between referring to external docs and rewriting them here. For ZCA, finding that balance is a greater challenge.

> Up until version 6, Plone has used server-side rendering to generate HTML-based content, with advanced resource management features for adding and bundling CSS and JavaScript. 

Reviewer 2: ```suggestion
Throughout its history, Plone has used server-side rendering to generate HTML-based content, with advanced resource management features for adding and bundling CSS and JavaScript. 
```

> Additionally, Plone's use of the ZCA makes it easy to extend and custosize, allowing users to create unique, feature-rich websites that are tailored to their specific needs.

Reviewer 2: ```suggestion
Additionally, Plone's use of the ZCA makes it easy to extend and customize, allowing users to create unique, feature rich websites that are tailored to their specific needs.
```

> Additionally, Plone's use of the ZCA makes it easy to extend and customize, allowing users to create unique, feature rich websites that are tailored to their specific needs.

Reviewer 3: I'd suggest removing the acronym, even though it was introduced above, to make it a bit easier to read for people who don't know it.

```suggestion
Additionally, Plone's use of a component architecture makes it easy to extend and customize, allowing users to create unique, feature rich websites that are tailored to their specific needs.
```

With the release of Plone 6, you now have the option to choose from two different out-of-the-box supported configurations when setting up a new Plone website.

> The Python-based backend server in Plone alone can still be used to render content server-side and deliver HTML to the browser, a setup that is referred to in the Plone documentation as "Classic UI". 

Reviewer 2: ```suggestion
The Python-based backend server in Plone can still be used alone to render content server-side and deliver HTML to the browser, a setup that is referred to in the Plone documentation as "Classic UI". 
```

> This configuration has been supported by Plone since its initial release and is still available in the latest versions of the platform. 

Reviewer 2: ```suggestion
This configuration has been supported by Plone since its initial release and is still available in the latest version of the platform. 
```

> For container-based deployment, only the plone-backend image is required, or a derivative image with customisations added. 

Reviewer 2: ```suggestion
For container-based deployment, only the `plone-backend` image is required, or a derivative image with customizations added. 
```

> For container-based deployment, only the `plone-backend` image is required, or a derivative image with customizations added. 

Reviewer 3: ```suggestion
For container-based deployment, this option requires only the `plone-backend` image, or a derivative image with customizations added. 
```

Reviewer 2: I think @davisagli's suggestion rephrases the opening clause and essentially repeats it.

I think it would be easier to split this into two sentences, one to establish the requirement of the image as a base, and one to say it may be modified.

```suggestion
For container-based deployment, only the `plone-backend` image is required.
It may be used as a base image, adding customizations, to make a derivative image.
```

Either this or the original works for me.

Author: I'll go with the 2 shorter sentences, was struggling with this from the beginnen how to succinctly point out that you can build upon the base image.

> The default and advised configuration for new websites in Plone is to use our new React Based javascript frontend called 'Volto'.

Reviewer 2: ```suggestion
The default and recommended configuration for new websites in Plone is the new React-based JavaScript frontend called "Volto".
```

> For this setup you still need to run the Python based backend server, but with the REST API enabled and an updated configuration profile.

Reviewer 2: ```suggestion
For this setup you still need to run the Python-based backend server, as well as enable the REST API, and update the configuration profile.
```

> For this setup you still need to run the Python-based backend server, as well as enable the REST API, and update the configuration profile.

Reviewer 3: ```suggestion
For this setup you still need to run the Python-based backend server, and update the configuration profile.
```

(It comes enabled, doesn't it?)

Reviewer: > (It comes enabled, doesn't it?)

yes.

Author: Added an extra sentence here to explain that these setup steps are done automaticaly if you select the correct site in the site creation form. We could add an image/screenshot later.

> In addition a separate NodeJS based frontend server will serve the javascript frontend resources and provide SSR with hydration.

Reviewer 2: ```suggestion
In addition a separate NodeJS based frontend server will serve the JavaScript frontend resources and provide SSR with hydration.
```

> To deploy this setup using containers you will need the plone-frontend image of the frontend server.

Reviewer 2: ```suggestion
To deploy this setup using containers, you will need the `plone-frontend` image for the frontend server.
```

> Plone 6 is the version Plone where we support two programming language stacks, one for Python and Javascript.

Reviewer 2: ```suggestion
Beginning with Plone 6, we now support two programming language stacks, one each for Python and JavaScript.
```

The documentation has been rewritten, but for this first release you will find some repetition of concepts in the documentation structure.

> For example for the development setup and information on deployment options.

Reviewer 2: ```suggestion
See, for example, the development setup and information on deployment options.
```

> It will take some time before we find and can implement the best structure to explain these new possibilities and expansion of Plone its capabilities.

Reviewer 2: ```suggestion
It will take some time until we find and can implement the best structure to explain these new possibilities and the expansion of Plone's capabilities.
```

```{seealso}
https://training.plone.org/mastering-plone/what_is_plone.html
```

(overview-5-label)=
## Deployment

> To run a public Plone website in production, you will also need to configure and run a reverse proxy (or ingress), arrange for SSL certificates (either from Let's Encrypt or manually), guarantee persistence of the content database and arrange for backups.

Reviewer 2: ```suggestion
To run a public Plone website in production, you will also need to configure and run a reverse proxy (or ingress), arrange for SSL certificates (either from Let's Encrypt or manually), guarantee persistence of the content database, and arrange backups.
```

> This is the domain of sysadmins and modern devops.

Reviewer 2: ```suggestion
This is the domain of systems administrators and modern developer-operations professionals.
```

> Our documentation contains setup examples for these services but requires also generic experience and knowledge of these domains.

Reviewer 2: ```suggestion
Our documentation contains setup examples for these services, yet requires that the reader have some generic experience and knowledge of these domains.
```

(overview-6-label)=
## Good to know / What to know

> One of the key benefits of the new React based frontend for Plone 6 is that you can now customize and theme Plone extensively using HTML and Javascript using up to date frontend technologies without having having to set up a local Python development environment.

Reviewer 2: ```suggestion
One of the key benefits of the new React-based frontend for Plone 6 is that you can now customize and theme Plone extensively using HTML, CSS, and JavaScript using up-to-date frontend technologies without having having to set up a local Python development environment.
```

The Plone backend can be run on a local developer machine in a container.

> Basic familiarity with programming in Python and managing Python modules/packages using virtualenv and pip is required to work on the backend code.

Reviewer 2: ```suggestion
Basic familiarity with programming in Python and managing Python modules and packages using `virtualenv` and `pip` is required to work on the backend code.
```

> We use virtualenv and mxdev to manage the source installation of packages in Plone 6.

Reviewer 2: ```suggestion
We use `virtualenv` and {term}`mxdev` to manage the source installation of packages in Plone 6.
```

> Similarly, to develop for the new React frontend, you need to have some experience with setting up NodeJS, using a tool like NVM (Node Version Manager) to isolate your setup, Yarn and React.

Reviewer 2: ```suggestion
Similarly, to develop for the new React frontend, you need to have some experience with setting up NodeJS, using a tool like NVM (Node Version Manager) to isolate your setup, and familiarity with {term}`Yarn` and {term}`React`.
```

> If you are looking for more study material on these technologies and the documentation  is too dense, see and follow one or more [Plone Trainings](https://training.plone.org).

Reviewer 2: ```suggestion
If you are looking for more study material on these technologies beyond the documentation, see and follow one or more [Plone Trainings](https://training.plone.org).
```

With the release of Plone 6, you now have the option to choose from two different out-of-the-box supported configurations when setting up a new Plone website.
The Python-based backend server in Plone can still be used alone to render content server-side and deliver HTML to the browser, a setup that is referred to in the Plone documentation as "Classic UI". 
This configuration has been supported by Plone since its initial release and is still available in the latest version of the platform.
For container-based deployment, only the `plone-backend` image is required.
It may be used as a base image, adding customizations, to make a derivative image.

The default and recommended configuration for new websites in Plone is the new React-based JavaScript frontend called "Volto".
For this setup you still need to run the Python-based backend server, as well as enable the REST API, and update the configuration profile.

> These settings and profile are applied automatically when you select the `Create Plone Site` option in the Plone website creation form.

Reviewer 2: I think that is a button control, correct?

```suggestion
These settings and profile are applied automatically when you select the {guilabel}`Create Plone Site` option in the Plone website creation form.
```

Author: Yes it is, thanks!
